packages/auth-providers/supabase/middleware/build.mts

```diff
@@ -1,3 +1,35 @@
-import { build } from '@redwoodjs/framework-tools'
+import {
+  build,
+  defaultBuildOptions,
+} from '@redwoodjs/framework-tools'
 
-await build()
+import { generateCjsTypes} from '@redwoodjs/framework-tools/cjsTypes'
+import { writeFileSync } from 'node:fs'
+
+// CJS build
+await build({
+  buildOptions: {
+    ...defaultBuildOptions,
+    outdir: 'dist/cjs',
+    packages: 'external',
+  },
+})
+
+// ESM build
+await build({
+  buildOptions: {
+    ...defaultBuildOptions,
+    format: 'esm',
+    packages: 'external',
+  },
+})
+
+// Place a package.json file with `type: commonjs` in the dist/cjs folder so that
+// all .js files are treated as CommonJS files.
+writeFileSync('dist/cjs/package.json', JSON.stringify({ type: 'commonjs' }))
+
+// Place a package.json file with `type: module` in the dist folder so that
+// all .js files are treated as ES Module files.
+writeFileSync('dist/package.json', JSON.stringify({ type: 'module' }))
+
+generateCjsTypes()
```

Comment:
I think you want to `await` this, right?

Reply:
Ah yeah.